lib/ansible/module_utils/urls.py

```diff
@@ -1128,8 +1128,6 @@ def open(self, method, url, data=None, headers=None, use_proxy=None,
         urllib_request.install_opener(opener)
 
         data = to_bytes(data, nonstring='passthru')
-        if method not in ('OPTIONS', 'GET', 'HEAD', 'POST', 'PUT', 'DELETE', 'TRACE', 'CONNECT', 'PATCH'):
-            raise ConnectionError('invalid HTTP request method; %s' % method)
         request = RequestWithMethod(url, method, data)
```

Comment:
Since we are no longer asserting that the provided method is allowed, and since we are already doing it in the module, I think we should be doing `method.upper()` here, for those people using this code directly, and not through `uri`.

Reply:
The requests library internally uses lowercase methods, so I don't think this really is a problem.